src/lesson9/HomemadeArrayList/hArrayList.java

```diff
@@ -0,0 +1,129 @@
+package lesson9.HomemadeArrayList;
+
+public class hArrayList implements hList {
+
+    private int[] array;
+
+    public hArrayList(int capacity) {
+        this.array = new int[capacity];
+    }
+
+    @Override
+    public int get(int index) {
+        return array[index];
+    }
+
+    @Override
+    public void set(int index, int item) {
+        array[index] = item;
+    }
+
+    @Override
+    public int size() {
+        return array.length;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size() == 0;
+    }
+
+    @Override
+    public boolean contains(int item) {
+        return indexOf(item) >= 0;
+    }
+
+    @Override
+    public int[] toArray() {
+        return array;
+    }
+
+    @Override
+    public void add(int item) {
+        int[] tempArr = array;
+        array = new int[tempArr.length + 1];
```

Comment:
not good idea to create new array each time when you want to add element, take a look at the imp of ArrayList. hint you can have var capacity